staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/strategy.go

```diff
@@ -405,7 +405,9 @@ func (a customResourceStrategy) MatchCustomResourceDefinitionStorage(label label
 func hasBlockingErr(errs field.ErrorList) (bool, *field.Error) {
 	for _, err := range errs {
 		if err.Type == field.ErrorTypeNotSupported || err.Type == field.ErrorTypeRequired || err.Type == field.ErrorTypeTooLong || err.Type == field.ErrorTypeTooMany || err.Type == field.ErrorTypeTypeInvalid {
-			return true, field.Invalid(nil, nil, "some validation rules were not checked because the object was invalid; correct the existing errors to complete validation")
+			berr := *err
```

Comment:
An alternative could be to use `ErrorTypeInternal` like so:
```go
return true, &field.Error{
    Type:   field.ErrorTypeInternal,
    Field:  err.Field,
    Detail: "some validation rules were not checked because the object was invalid; correct the existing errors to complete validation",
}
```
Can not use `InternalError` constructor as it requires *Path that is not available at this point:
https://github.com/kubernetes/kubernetes/blob/f3cbd79db7f0c86a2d3602fdff6b174543d2cf1c/staging/src/k8s.io/apimachinery/pkg/util/validation/field/errors.go#L265-L270

Reply:
It's on purpose that we return invalid error here. The main idea is that in certain cases, it doesn't make a lot of sense to continue cel validation if there is already certain error existing in validation. Hence we are gonna return invalid error and let user clear up other errors before continue validating.

Reply:
Sure, makes sense. This change just adds a bit of details to the returned error, it does not change the error type.